docs/site/Advanced-sequence-topics.md

```diff
@@ -0,0 +1,320 @@
+namespace.
+
+## Query string parameters and path parameters
+
+OAI 3.0.x describes the data from a request’s header, query and path in an
+operation specification’s parameters property. In a Controller method, such an
+argument is typically decorated by `@param()`. We've made multiple shortcuts
+available to the `@param()` decorator in the form of
+`@param.<http_source>.<OAI_primitive_type>`. Using this notation, path
+parameters can be described as `@param.path.string`. Here is an example of a
+controller method which retrieves a Note model instance by obtaining the `id`
+from the path object.
+
+```ts
+@get('/notes/{id}', {
+  responses: {
+    '200': {
+      description: 'Note model instance',
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Note, {includeRelations: true}),
+        },
+      },
+    },
+  },
+})
+async findById(
+  @param.path.string('id') id: string,
+  @param.filter(Note, {exclude: 'where'}) filter?: FilterExcludingWhere<Note>
+): Promise<Note> {
+  return this.noteRepository.findById(id, filter);
+}
+```
+
+(Notice: the filter for `findById()` method only supports the `include` clause
+for now.)
+
+You can also specify a parameter which is an object value encoded as a JSON
+string or in multiple nested keys. For a JSON string, a sample value would be
+`location={"lang": 23.414, "lat": -98.1515}`. For the same `location` object, it
+can also be represented as `location[lang]=23.414&location[lat]=-98.1515`. Here
+is the equivalent usage for `@param.query.object()` decorator. It takes in the
+name of the parameter and an optional schema or reference object for it.
+
+```ts
+@param.query.object('location', {
+  type: 'object',
+  properties: {lat: {type: 'number', format: 'float'}, long: {type: 'number', format: 'float'}},
+})
+```
+
+The parameters are retrieved as the result of `parseParams` Sequence action.
+Please note that deeply nested properties are not officially supported by OAS
+yet and is tracked by
+[OAI/OpenAPI-Specification#1706](https://github.com/OAI/OpenAPI-Specification/issues/1706).
+Therefore, our REST API Explorer does not allow users to provide values for such
+parameters and unfortunately has no visible indication of that. This problem is
+tracked and discussed in
+[swagger-api/swagger-js#1385](https://github.com/swagger-api/swagger-js/issues/1385).
+
+## Parsing Requests
+
+Parsing and validating arguments from the request url, headers, and body. See
+page [Parsing requests](Parsing-requests.md).
+
+## Invoking controller methods
+
+The `invoke` sequence action simply takes the parsed request parameters from the
+`parseParams` action along with non-decorated arguments, calls the corresponding
+controller method or route handler method, and returns the value from it. The
+default implementation of
+[invoke](https://github.com/strongloop/loopback-next/blob/6bafa0774662991199090219913c3dc77ad5b149/packages/rest/src/providers/invoke-method.provider.ts)
```

Comment:
Perhaps we should link to the latest code? That way, it's obvious if it's outdated. Otherwise, users may be unknowingly reading an outdated version of the codebase.

Perhaps the API Docs should contain the source code snippet too, so that there's a more "permanent" link to the latest code; But that would be a separate discussion.

```suggestion
[invoke](https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/providers/invoke-method.provider.ts)
```